pkg/operator/status/status_controller.go

```diff
@@ -131,6 +130,19 @@ func (c StatusSyncer) sync() error {
 		return nil
 	}
 
+	// When operator is not allowed to be unmanagable, but the state is set to unmanagable, prevent upgrades done by this operators (but do not prevent managing operand).
+	if IsOperatorAlwaysManaged() && detailedSpec.ManagementState == operatorv1.Unmanaged {
+		clusterOperatorObj.Status = configv1.ClusterOperatorStatus{}
+		configv1helpers.SetStatusCondition(&clusterOperatorObj.Status.Conditions, configv1.ClusterOperatorStatusCondition{Type: configv1.OperatorUpgradeable, Status: configv1.ConditionUnknown, Reason: "Unmanaged"})
+
+		if !equality.Semantic.DeepEqual(clusterOperatorObj, originalClusterOperatorObj) {
+			if _, updateErr := c.clusterOperatorClient.ClusterOperators().UpdateStatus(clusterOperatorObj); err != nil {
```

Comment:
This will be overwritten below by the union upgradeable, right?

Reply:
good catch, missed return nil, fixed.